artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -558,6 +564,21 @@ public void physicalSend(Command command) throws IOException {
 
    }
 
+   private void chunkSend(final ByteSequence bytes, final int bufferSize, final int maxPacketSize) {
+      if (logger.isTraceEnabled()) {
+         logger.trace("Sending a big packet sized as {} with smaller packets of {}", bufferSize, maxPacketSize);
+      }
+      for (int posBuffer = 0; posBuffer < bufferSize; posBuffer += maxPacketSize) {
+         int chunkSize = Math.min(bufferSize - posBuffer, maxPacketSize);
+         if (logger.isTraceEnabled()) {
+            logger.trace("Sending a partial packet of {} bytes, starting at {}", chunkSize, posBuffer);
+         }
+         final ActiveMQBuffer buffer = transportConnection.createTransportBuffer(chunkSize);
```

Comment:
can that one be reused? should always of the right size or smaller

Reply:
I'm splitting a big buffer chunk into smaller buffers


so, if you have a 200MB message, I'm calling multiple writes instead of just one mega-write

Reply:
ah... no.. it can't be reused.... netty will be holding it down the pipeline...

Reply:
although I'm calling flush... but I would prefer to keep it through the pool like it's always done.


if I wanted to reuse it.. I would have to increase the the counter with a refUp here.. I would prefer to not deal with that.

Reply:
I tried to make a change with the same buffer, but I got some refCounts issues... I would prefer to keep the way it is now.

Reply:
makes sense, it can't be in the pipeline twice.